merger.js

```diff
@@ -0,0 +1,161 @@
+        return null;
+    }
+
+    return opts;
+}
+
+program.action((options) => {
+    const opts = validateOptions(options);
+    if (opts === null)
+        process.exit(1);               
+
+    const dirPath = opts.outputDir;
+
+    try {
+        fs.accessSync(dirPath, fs.constants.R_OK);
+    } catch (err) {
+        process.stderr.write(
+            `ERR: The directory '${opts.outputDir}' doesn't exist or permissions aren't set correctly.\n`
+        );
+        process.exit(1);
+    }
+
+    const stats = { tests: 0, failures: 0, skipped: 0 };
+    const suites = [];
+    const resultFiles = fs.readdirSync(dirPath).filter((file) => {
+        return file.includes('tmp');
+    });
+
+    for (const file of resultFiles) {
+        const data = fs.readFileSync(`${dirPath}/${file}`, 'utf8', (err, data) => {
+            if (err) {
+                process.stderr.write(`ERR: Cannot read contents of ${file}.\n`);
+                process.exit(1);
+            }
+
+            return data;
+        });
+
+        const xmlDoc = convert.xml2json(data, { compact: true, spaces: 2 });
+        let suite = JSON.parse(xmlDoc);
+
+        for (const [key, value] of Object.entries(suite.testsuites)) {
+            let file;
+            if (key === '_attributes') {
+                stats.tests += parseInt(value['tests']);
+                stats.failures += parseInt(value['failures']);
+                stats.skipped += parseInt(value['skipped']);
+                continue;
+            }
+            for (const [k, v] of Object.entries(value)) {
+                if (v['_attributes']['file'] != 'null') 
+                    file = v['_attributes']['file'];
+                else 
+                    v['_attributes'].file = file;
+
+                if (v['testcase']) 
+                    suites.push({ ...v });
+            }
+        }
+    }
+
+    const source = opts.source ?? process.env.BUILD_TAG ?? process.env.RUNNER_TRACKING_ID ?? 'local';
+
+    const propList = [
+        { _attributes: { key: 'project', value: opts?.project ?? 'sample-project' } },
+        { _attributes: { key: 'component', value: opts?.component ?? 'sample-component' } },
+        { _attributes: { key: 'source', value: source } },
+        { _attributes: { key: 'env', value: opts?.env ?? '' } },
+    ];
+
+    const propObj = {property: [...propList]}
+
+    const rootSuite = {
+        _attributes: { tests: stats.tests, failures: stats.failures, errors: 0, skipped: stats.skipped },
+        properties: [propObj],
+        testsuite: suites,
+    };
+    const out = { testsuites: [rootSuite] };
+
+    const data = convert.json2xml(JSON.stringify(out), { compact: true, ignoreComment: true, spaces: 2 });
+    
+    if (opts.verbose) 
+        console.log(data);
+    
+    try {
+        fs.writeFileSync(`${dirPath}/${opts?.output ?? 'merged'}.ibutsu.xml`, data, 'utf-8');
+    } catch (exc) {
+        process.stderr.write('ERR: Cannot write results file.\n');
```

Comment:
You probably want to see if you can pull the error message out of the exception here, so that people have more information about why the file write is failing, as opposed to a generic and unhelpful single message for all errors.